src/Microsoft.DotNet.Build.Tasks/VersionTools/MsBuildTraceListener.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks.VersionTools
+{
+    public class MsBuildTraceListener : TraceListener
+    {
+        private TaskLoggingHelper _log;
+        private TraceEventType _eventType;
+        private StringBuilder _partialLine = new StringBuilder();
+
+        public MsBuildTraceListener(TaskLoggingHelper log, TraceEventType eventType)
+        {
+            _log = log;
+            _eventType = eventType;
+
+            Filter = new TraceEventTypeFilter
+            {
+                ShouldTraceType = eventType
+            };
+        }
+
+        public override void Write(string message)
+        {
+            _partialLine.Append(message);
```

Comment:
The way I use `Trace`, `Write` is only ever used by `TraceListener` internals to print a prefix header like "msbuild Information: 0 :". It's possible someone will use `Trace.Write` later, so I'll fix this by adding some kind of `Flush`.

---
In reply to: [69950623](https://github.com/dotnet/buildtools/pull/863#discussion_r69950623) [](ancestors = 69950623)